### ISSUE TITLE
Yarn install --offline fails for scoped package with file resolution override

### ISSUE THREAD

**Issue by gpeal** [yarnpkg/yarn#4909]
**What is the current behavior?**
1) Set up a project with a local tarball mirror and pruning ([Sample Project](https://github.com/yarnpkg/yarn/files/1465015/yarn-test.zip)).
2) Set the resolution for a scoped package to a file in package.json.
3) Clear the global yarn cache.
4) Run a `yarn install --offline`.
Yarn will first try and resolve the package to the tarball which will fail. It will then look in the global cache but the scoped cache dir (in this case `npm-@types`) doesn't exist and the install will fail [here](https://github.com/yarnpkg/yarn/blob/1c845bd2dc64fc4937d82ac7b88d014e9fd6b3bb/src/resolvers/registries/npm-resolver.js#L109).

This seems like a very niche use case but our CI machines do a build with this exact setup so we can't use file resolutions until this is resolved.

@lelandrichardson

**Comment by JeremyJonas** [comment]
Looks to be same as #731

**Comment by jphilipstevens** [comment]
Any update on this issue?

**Comment by stevenroose** [comment]
Yeah so I think I'm having this issue. Yarn.lock has an entry like this

```
http-signature@1.3.4, http-signature@~1.2.0:
  version "1.3.4"
```

Which seems to say that version 1.3.4 is selected. But then when you use `--offline` it will still look for a version that satisfies `~1.2.0` (which `1.3.4` doesn't) and complain that the version doesn't exist.

So either yarn intends to force packages that depend on `~1.2.0` to use `1.3.4` instead and is failing to do that, or it is omitting certain version because of the presence of "resolutions".
